src/Routes/Devices/RetryUpdatePopover.js

```diff
@@ -0,0 +1,163 @@
+  DescriptionListDescription,
+  Stack,
+  StackItem,
+} from '@patternfly/react-core';
+import { DescriptionListTermHelpText } from '@patternfly/react-core';
+import ExclamationCircleIcon from '@patternfly/react-icons/dist/esm/icons/exclamation-circle-icon';
+import { DateFormat } from '@redhat-cloud-services/frontend-components/DateFormat';
+import apiWithToast from '../../utils/apiWithToast';
+import { updateDeviceLatestImage } from '../../api/devices';
+import { useDispatch } from 'react-redux';
+import PropTypes from 'prop-types';
+
+function getDevicePopoverDescription(props) {
+  if (props.device.DispatcherReason === 'The playbook failed to run.') {
+    return (
+      <div>
+        The playbook failed to run. You can retry the update or build a new one.
+        <Stack className="pf-u-mt-sm">
+          <StackItem className="pf-u-font-weight-bold">Last Seen</StackItem>
+          <StackItem> {<DateFormat date={props.lastSeen} />}</StackItem>
+        </Stack>
+      </div>
+    );
+  }
+  if (
+    props.device.DispatcherReason ===
+    'The service timed out during the last update.'
+  ) {
+    return (
+      <div>
+        The service timed out during the last update. You can retry the update
+        or build a new one.
+        <Stack className="pf-u-mt-sm">
+          <StackItem className="pf-u-font-weight-bold">Last Seen</StackItem>
+          <StackItem> {<DateFormat date={props.lastSeen} />}</StackItem>
+        </Stack>
+      </div>
+    );
+  }
+
+  if (props.device.DispatcherStatus === 'UNRESPONSIVE') {
+    return (
+      <div>
+        The service could not be reached via RHC. The device may communicate at
+        a later time if this is a network issues or could be an indication of a
+        larger problem.
+        <Stack className="pf-u-mt-sm">
+          <StackItem className="pf-u-font-weight-bold">Last Seen</StackItem>
+          <StackItem> {<DateFormat date={props.lastSeen} />}</StackItem>
+        </Stack>
+      </div>
+    );
+  }
+}
+
+function getDevicePopoverTitle(props) {
+  if (props.device.DispatcherReason === 'The playbook failed to run.') {
+    return <span className="pf-u-ml-xs"> Playbook error </span>;
+  }
+
+  if (
+    props.device.DispatcherReason ===
+    'The service timed out during the last update.'
+  ) {
+    return <span className="pf-u-ml-xs"> Service timed-out </span>;
+  }
+
+  if (props.device.DispatcherStatus === 'UNRESPONSIVE') {
+    return <span className="pf-u-ml-xs"> Unresponsive </span>;
+  }
+}
+
+const RetryUpdatePopover = (props) => {
+  const [isVisible, setIsVisible] = React.useState(false);
+  const dispatch = useDispatch();
+  props;
+  const isMounted = React.useRef(true);
+  React.useEffect(() => {
+    if (isMounted.current) {
+      return () => {
+        isMounted.current = false;
+      };
+    }
+  });
+  props.DeviceUUID;
+  const setUpdateModal = { setUpdateModal };
+  const statusMessages = {
+    onSuccess: {
+      title: 'Success',
```

Comment:
@mayleighnmyers: 

title and variant needs to be updated as well

```js
variant: 'info',
title: 'Updating device',
```

Similar to https://github.com/RedHatInsights/edge-frontend/pull/533/files#r972046811